core/shared/src/main/scala/com/monovore/decline/CommandApp.scala

```diff
@@ -22,7 +22,7 @@ class CommandApp(command: Command[Unit]) {
   }
 
   def main(args: Array[String]): Unit =
-    command.parse(args) match {
+    command.parse(PlatformApp.ambientArgs getOrElse args) match {
```

Comment:
Huh, interesting. The [intended use](http://ben.kirw.in/decline/usage.html#defining-an-application) of `CommandApp` is something like:

```scala
// in MyApp.scala
object MyApp extends CommandApp(
  name = "xxx",
  header = "xxx",
  main = {
    val params: Opts[XXX] = ???
    def run(xxx: XXX): Unit = ???
    params.map(run)
  }
}
```

...which, on the latest ScalaJS, would generate a script such that you could:

```bash
nodejs js/target/scala-2.11/decline-test-fastopt.js --my-param
```

The generated main method isn't intended to be called by user code... it's there so that the JVM / ScalaJS can use it as the entry point for the application. (I could probably document this better!) Would the above style work for your application, or is there some reason you can't structure your code this way?